snuba/pipeline/settings_delegator.py

```diff
@@ -25,7 +26,7 @@ class RateLimiterDelegate(RequestSettings):
 
     def __init__(self, prefix: str, delegate: RequestSettings):
         self.referrer = delegate.referrer
-        self.__delegate = delegate
+        self.__delegate = copy.deepcopy(delegate)
```

Comment:
great find on the race condition. I have a few questions:

1. Is this affecting customer facing queries right now?
2. This will have a performance impact as we will be deepcopying a lot of things (including the whole parsed query). Do we need to copy the whole request or just isolate the request settings?

Reply:
At the very least, put a comment here explaining why this deepcopy is done so nobody thinks "oh this seems like I could get a great performance impact if I removed it"

Reply:
> great find on the race condition. I have a few questions:
> 
> 1. Is this affecting customer facing queries right now?
> 2. This will have a performance impact as we will be deepcopying a lot of things (including the whole parsed query). Do we need to copy the whole request or just isolate the request settings?

1. Yes it is impacting customer facing queries. But the effects have been mitigated by increasing the quotas for now.
2. This does not copy the Request object(which has the parsed query). It only deepcopies the RequestSettings object which would have fields like https://github.com/getsentry/snuba/blob/master/snuba/request/request_settings.py#L95-L106

Reply:
Context on the user impact. 
This was most likely the root cause of the increase in rate limiting of post process queries. The spiky load theory may have been wrong, so we should get back to Dan as he does not need to introduce anything new and maybe wants to move the queries back to consistent mode.